tsdb/engine/tsm1/compact_test.go

```diff
@@ -2853,21 +2860,82 @@ func TestDefaultPlanner_FullyCompacted_ManySingleGen2GBLastLevel2(t *testing.T)
 	}
 
 	_, cgLen := cp.PlanLevel(1)
-	require.Equal(t, int64(0), cgLen, "compaction group length; PlanLevel(1)")
+	require.Zero(t, cgLen, "compaction group length; PlanLevel(1)")
 	_, cgLen = cp.PlanLevel(2)
-	require.Equal(t, int64(0), cgLen, "compaction group length; PlanLevel(2)")
+	require.Zero(t, cgLen, "compaction group length; PlanLevel(2)")
 	_, cgLen = cp.PlanLevel(3)
-	require.Equal(t, int64(0), cgLen, "compaction group length; PlanLevel(3)")
+	require.Zero(t, cgLen, "compaction group length; PlanLevel(3)")
 
-	_, cgLen = cp.Plan(time.Now().Add(-1))
-	require.Equal(t, int64(0), cgLen, "compaction group length; Plan()")
+	tsmP, pLenP := cp.Plan(time.Now().Add(-time.Second))
+	require.Zero(t, len(tsmP), "compaction group; Plan()")
+	require.Zero(t, pLenP, "compaction group length; Plan()")
 
 	tsm, cgLen, genLen := cp.PlanOptimize()
 	require.Equal(t, int64(1), cgLen, "compaction group length")
 	require.Equal(t, int64(3), genLen, "generation count")
 	require.Equal(t, len(expFiles), len(tsm[0]), "tsm files in compaction group")
 }
 
+// This test will check to ensure that any TSM generations planned with the default planner
+// using Plan() and PlanLevel() over default block size are skipped.
```

Comment:
How about a test of the scenario where you have default max block size for some files, aggressive max block size for some files, and a few new file with less than default max block size and less than max file size?

I'm thinking of a shard that was cold and fully compacted, then took some backfills.  What should we do there?

Reply:
Added `TestDefaultPlanner_BackfillMock` which incorporates a mixture of levels, block counts, file sizes, and generations. That should mimic a backfill sufficiently.